eng/common/scripts/Verify-RestApiSpecLocation.ps1

```diff
@@ -0,0 +1,281 @@
+<#
+.SYNOPSIS
+  Verifies the location of the REST API specification.
+
+.DESCRIPTION
+  This script is used to verify the REST API specifications used to generate SDK package are from the main branch of Azure/azure-rest-api-specs repository.
+
+.PARAMETER ServiceDirectory
+  The directory path of the service.
+
+.PARAMETER PackageName
+  The name of the package.
+
+.PARAMETER ArtifactLocation
+  The location of the generated artifact for the package.
+
+.PARAMETER GitHubPat
+  The GitHub personal access token used for authentication. By default, it will use the environment variable GH_TOKEN.
+
+.PARAMETER PackageInfoDirectory
+  The directory path where the package information is stored.
+
+.EXAMPLE
+  Verify-RestApiSpecLocation -ServiceDirectory "/home/azure-sdk-for-net/sdk/serviceab" -PackageName "MyPackage" -ArtifactLocation "/home/ab/artifacts" -GitHubPat "xxxxxxxxxxxx" -PackageInfoDirectory "/home/ab/artifacts/PackageInfo"
+
+#>
+[CmdletBinding()]
+param (
+  [Parameter(Position = 0)]
+  [ValidateNotNullOrEmpty()]
+  [string] $ServiceDirectory,
+  [Parameter(Position = 1)]
+  [ValidateNotNullOrEmpty()]
+  [string] $PackageName,
+  [ValidateNotNullOrEmpty()]
+  [string] $ArtifactLocation,
+  [Parameter(Position = 2)]
+  [string]$GitHubPat = $($env:GH_TOKEN),
```

Comment:
Get the token from environment variable so that it won't be leaked in the pipeline log.